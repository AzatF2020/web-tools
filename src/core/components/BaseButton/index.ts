export { default as BaseButton } from './ui/BaseButton.tsx';
