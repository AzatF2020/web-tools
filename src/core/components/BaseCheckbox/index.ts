export { default as BaseCheckbox } from './ui/BaseCheckbox';
