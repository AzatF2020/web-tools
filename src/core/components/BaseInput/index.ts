export type { IBaseInputProps } from './types/types';

export { default as BaseInput } from './ui/BaseInput.tsx';
