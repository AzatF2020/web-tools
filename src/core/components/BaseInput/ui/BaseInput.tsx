import { forwardRef } from 'react';
import type { IBaseInputProps } from '@/core/components/BaseInput/types/types';
import styles from '@/core/components/BaseInput/styles/BaseInput.module.scss';
import clsx from 'clsx';

const BaseInput = forwardRef<HTMLInputElement, IBaseInputProps>(
  ({ type = 'text', size = 'md', invalid = false, className, onChange, ...props }, ref) => {
    return (
      <input
        ref={ref}
        type={type}
        onChange={onChange}
        className={clsx(
          styles.input,
          styles[`input--${size}`],
          invalid && styles['input--invalid'],
          className,
        )}
        {...props}
      />
    );
  },
);

export default BaseInput;
