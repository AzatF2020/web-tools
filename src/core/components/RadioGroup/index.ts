export type { IRadioItemProps } from './types/types';

export { default as RadioItem } from './ui/RadioItem';
