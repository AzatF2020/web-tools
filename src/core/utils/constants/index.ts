const sizes = ['sm', 'md', 'lg'] as const;
const variants = ['primary', 'secondary', 'link', 'outlined'];

export { sizes, variants };
