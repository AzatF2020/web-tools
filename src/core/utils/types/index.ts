type TSize = 'sm' | 'md' | 'lg';

type TButtonVariant = 'primary' | 'secondary' | 'link' | 'outlined' | 'loading';

export type { TSize, TButtonVariant };
