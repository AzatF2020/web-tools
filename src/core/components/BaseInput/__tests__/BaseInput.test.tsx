import { describe, it, expect, vi } from 'vitest';
import { fireEvent, render } from '@testing-library/react';
import { BaseInput } from '@/core/components/BaseInput';
import { sizes } from '@/core/utils/constants';
import type { IBaseInputProps } from '@/core/components/BaseInput/types/types';
import type { TBaseInputSize } from '@/core/models/BsseInputModel/types';

const renderInput = (props: IBaseInputProps = { placeholder: 'Write text' }) => {
  return render(<BaseInput {...props} />);
};

describe('BaseInput test component', () => {
  const placeholder = 'Write text';

  describe('test prop size', () => {
    test.each(sizes)('should render component with size %s', (size) => {
      const { getByPlaceholderText } = renderInput({ size: size as TBaseInputSize, placeholder });
      expect(getByPlaceholderText(placeholder)).toHaveClass(`input--${size}`);
    });
  });

  it('should render component with default props', () => {
    const { getByPlaceholderText } = renderInput({
      placeholder,
      size: 'md' as TBaseInputSize,
      type: 'text',
      name: 'text',
      value: 'text for test',
      disabled: false,
      readOnly: false,
      invalid: false,
      onChange: vi.fn(),
    });

    expect(getByPlaceholderText('Write text')).toMatchSnapshot();
  });

  it('should render with incorrect prop', () => {
    const { getByPlaceholderText } = renderInput({
      placeholder,
      size: 'md' as TBaseInputSize,
    });

    expect(getByPlaceholderText(placeholder)).not.toHaveClass('input--sm');
  });

  it('should render with disabled or readonly attribute', () => {
    const { getByPlaceholderText } = renderInput({
      placeholder,
      disabled: true,
      readOnly: true,
    });

    const input = getByPlaceholderText(placeholder);

    expect(input).toBeDisabled();
    expect(input).toHaveAttribute('readonly');
  });

  it('should test onChange function', async () => {
    const onChange = vi.fn();
    const { getByPlaceholderText } = renderInput({ onChange, placeholder });

    await fireEvent.change(getByPlaceholderText(placeholder), {
      target: { value: 'value for test' },
    });

    expect(onChange).toHaveBeenCalledOnce();
  });

  it('should test onChange function with correct value', async () => {
    const { getByPlaceholderText } = renderInput();

    const input = getByPlaceholderText(placeholder);
    await fireEvent.change(input, { target: { value: 'value for test' } });
    expect(input).toHaveValue('value for test');
  });
});
