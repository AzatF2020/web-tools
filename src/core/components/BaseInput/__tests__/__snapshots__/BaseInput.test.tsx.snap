// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput test component > should render component with default props 1`] = `
<input
  class="input input--md"
  name="text"
  placeholder="Write text"
  type="text"
  value="text for test"
/>
`;
